python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -1008,10 +1012,19 @@ def build_asset_outs(asset_outs: Mapping[str, AssetOut]) -> Mapping[AssetKey, Tu
     return outs_by_asset_key
 
 
-def _make_asset_keys(deps: Optional[Set[CoercibleToAssetKey]]) -> Optional[Set[AssetKey]]:
+def _make_asset_keys(
+    deps: Optional[Set[Union[CoercibleToAssetKey, AssetsDefinition]]]
+) -> Optional[Set[AssetKey]]:
     """Convert all str items to AssetKey in the set."""
     if deps is None:
         return deps
 
-    deps_asset_keys = {AssetKey.from_coercible(dep) for dep in deps}
+    deps_asset_keys: Set[AssetKey] = set()
+    for dep in deps:
+        if isinstance(dep, AssetsDefinition):
+            # this assumes that if you pass a multi-asset that the downstream depends on all assets in the upstream multi-asset
+            deps_asset_keys.update(dep.keys)
```

Comment:
we could potentially try to use more information from the assets definition here, but i'm not sure what we need just yet